concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -466,53 +470,55 @@ checkForProtocolUpdate = liftSkov body
+                            case Vec.last cfgs of
+                                EVersionedConfiguration vc -> return (vcGenesisHeight vc)
+                        let vcIndex = fromIntegral (length existingVersions)
+                        -- construct the the new skov instance
+                        let vcGenesisHeight = 1 + localToAbsoluteBlockHeight latestEraGenesisHeight pvInitFinalHeight
+                        let newGSConfig =
+                                ( Skov.SkovConfig @newpv @gc @fc
+                                    ( globalStateConfig
+                                        (mvcStateConfig mvConfiguration)
+                                        (mvcRuntimeParameters mvConfiguration)
+                                        vcIndex
+                                        vcGenesisHeight
+                                    )
+                                    (mvcFinalizationConfig mvConfiguration)
+                                    UpdateHandler
+                                )
+                        -- clear data we no longer need after the protocol update
+                        Skov.clearSkovOnProtocolUpdate
+                        -- migrate the final block state into the new skov instance, and establish
+                        -- all the necessary transaction table, and other, invariants.
+                        (vcContext, st) <- Skov.SkovT $ \_ ctx -> do
+                            currentState <- State.get
+                            liftIO $
+                                runLoggerT
+                                    (Skov.migrateExistingSkov ctx currentState pvInitMigration nextGenesis newGSConfig)
+                                    mvLog
+                        -- Close down and resources that the old instance retains. We do this after
+                        -- since, e.g., caches and the transaction table are needed during migration.
+                        Skov.terminateSkov
+                        -- wrap up, notify the network layer, and add the new instance to
+                        -- the end of the mvVersions list
+                        liftIO $ do
+                            vcState <- liftIO $ newIORef st
+                            let vcShutdown = Skov.shutdownSkov vcContext =<< liftIO (readIORef vcState)
+                            let newEConfig :: VersionedConfiguration gc fc newpv
+                                newEConfig = VersionedConfiguration{..}
+                            writeIORef mvVersions (existingVersions `Vec.snoc` newVersion newEConfig)
+                            -- Notify the network layer we have a new genesis.
+                            let Callbacks{..} = mvCallbacks
+                            liftIO $ notifyRegenesis (Just (regenesisBlockHash nextGenesis))
+                            return ()
+                    SConsensusParametersVersion1 -> error "New consensus version not implemented yet." -- TODO: implement
```

Comment:
Same here.

Reply:
Added this task: https://github.com/Concordium/concordium-node/issues/700